### DISCUSSION TITLE
Button with square icon have width bigger than parent(but not height), bug or not?

### DISCUSSION THREAD

**Discussion by qarmin** [slint-ui/slint#3981] [Q&A]
```
import { VerticalBox, ProgressIndicator, Button } from "std-widgets.slint";

component Cor inherits HorizontalLayout {
    width: 20px;
    height: 20px;
    Button {
        horizontal-stretch: 1.0;
        icon: @image-url("../icons/settings.svg");
        clicked => {
            debug("clicked");
        }
    }
}

export component Exter inherits Window {
    width: 400px;
    height: 400px;
    Rectangle {
        background: red;
        Rectangle {
            width: 20px;
            height: 20px;
            background: blue;
            Cor {
                
            }
        }
        
    }
}
```
and icon 
[settings.svg.zip](https://github.com/slint-ui/slint/files/13430467/settings.svg.zip)

draw such items - is this expected or this is a bug?
Quite strange that only witdth is exceeded, because image is square
![Screenshot from 2023-11-21 20-54-41](https://github.com/slint-ui/slint/assets/41945903/7e3d69f2-7a18-4ab8-a9ed-b9099f7cbba9)

**Comment by ogoffart** [comment]
I think the issue here is that the button defines a min-width and so the layout constraints cannot be resolved.

**Comment by ghost** [comment]
The problem e.g. for the fluent button on this example. We have a min-width of 32px. And on icon only we have 20px width for the icon, left-padding of 12px and right-padding also 12px (regarding to the fluent design guidelines) what makes 44px. 

But I think on default icon only buttons are square. So maybe if no text is set we can adjust the padding that e.g. the fluent button has a size of 32px * 32px. What do you think?
